tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -880,19 +891,25 @@ final class ScalaJSClassEmitter private (
         envField("m", cd.name.name)
     }
 
+    genClassModuleExportDef(cd, tree.fullName, baseAccessor)
+  }
+
+  private def genClassModuleExportDef(cd: LinkedClass, exportFullName: String,
+      exportedValue: js.Tree)(implicit pos: Position): js.Tree = {
```

Comment:
Also, consider dropping `export` from `exportFullName`. IIUC there is no other name.